### PR TITLE
Add cmake flag to force use of system fastdds

### PR DESCRIPTION
**Pull request by Acuadros95**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,7 @@ option(UAGENT_ISOLATED_INSTALL "Install the project and dependencies into separa
 option(UAGENT_USE_INTERNAL_GTEST "Enable internal GTest libraries." OFF)
 option(BUILD_SHARED_LIBS "Control shared/static building." ON)
 
+option(UFORCE_SYSTEM_FAST "Find and use system FastMiddleware." ON)
```
Comment (pablogs9) [inline CMakeLists.txt:30]:
```suggestion
option(UAGENT_USE_SYSTEM_FASTDDS "Find and use system FastMiddleware." OFF)
```

File: cmake/SuperBuild.cmake [hunk 2]
```diff
@@ -77,7 +77,7 @@ if(NOT fastcdr_FOUND)
     list(APPEND _deps fastcdr)
 endif()
 
-if(UAGENT_FAST_PROFILE)
+if(UAGENT_FAST_PROFILE AND NOT UFORCE_SYSTEM_FAST)
```
Comment (pablogs9) [inline cmake/SuperBuild.cmake:80]:
```suggestion
if(UAGENT_FAST_PROFILE AND NOT UAGENT_USE_SYSTEM_FASTDDS)
```

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Linux/369/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Windows/370/)

**Comment by pablogs9** [comment]
Tests failing because `UFORCE_SYSTEM_FAST` defaulted to ON on a system with no Fast-DDS installation.

File: CMakeLists.txt [hunk 3]
```diff
@@ -77,10 +78,14 @@ if(UAGENT_P2P_PROFILE)
 endif()
 
 if(UAGENT_FAST_PROFILE)
-    set(_fastdds_version 2.0.1)
-    set(_fastdds_tag 2.0.x)
+    if(UAGENT_FAST_PROFILE AND NOT AGENT_USE_SYSTEM_FASTDDS)
```
Comment (pablogs9) [inline CMakeLists.txt:81]:
```suggestion
    if(UAGENT_FAST_PROFILE AND NOT UAGENT_USE_SYSTEM_FASTDDS)
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -77,10 +78,14 @@ if(UAGENT_P2P_PROFILE)
...
+        set(_fastdds_version 2.0.0)
```
Comment (pablogs9) [inline CMakeLists.txt:82]: Are we sure that setting `_fastdds_version` to  `2.0.0` will look for the minimum version [here](https://github.com/eProsima/Micro-XRCE-DDS-Agent/blob/38de6ecc739b9104bea5a582bbabb51cd4adbbb7/CMakeLists.txt#L135)?

I see [here](https://cmake.org/cmake/help/v3.0/command/find_package.html) that the format is: `major[.minor[.patch[.tweak]]]` so maybe we should use `set(_fastdds_version 2)`. Can you test it locally before fixing it in the PR?

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Linux/372/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Windows/373/)

File: CMakeLists.txt [hunk 5]
```diff
@@ -77,10 +78,14 @@ if(UAGENT_P2P_PROFILE)
...
 
 if(UAGENT_FAST_PROFILE)
-    set(_fastdds_version 2.0.1)
-    set(_fastdds_tag 2.0.x)
+    if(UAGENT_FAST_PROFILE AND NOT UAGENT_USE_SYSTEM_FASTDDS)
+        set(_fastdds_version 2.0.0)
+    else()
+        set(_fastdds_version 2.0.2)
+        set(_fastdds_tag v2.0.2)
+        set(_foonathan_memory_tag c619113) # This tag should be updated every time it gets updated in foonathan_memory_vendor eProsima's package
```
Comment (pablogs9) [inline CMakeLists.txt:86]: This should be out of this `if` statement. And also check that the if statement in line 80 is duplicated.

File: CMakeLists.txt [hunk 6]
```diff
@@ -77,10 +78,14 @@ if(UAGENT_P2P_PROFILE)
 endif()
 
 if(UAGENT_FAST_PROFILE)
-    set(_fastdds_version 2.0.1)
-    set(_fastdds_tag 2.0.x)
+    if(UAGENT_FAST_PROFILE AND NOT UAGENT_USE_SYSTEM_FASTDDS)
```
Comment (pablogs9) [inline CMakeLists.txt:81]:
```suggestion
    if(NOT UAGENT_USE_SYSTEM_FASTDDS)
```

File: CMakeLists.txt (hunk 5 above)
Comment (pablogs9) [reply CMakeLists.txt:86]: My fault.

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Linux/374/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Windows/375/)

**Review by pablogs9 — APPROVED** [review]

**Review by jamoralp — APPROVED** [review]

**Pull request merged**